app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt

```diff
@@ -0,0 +1,120 @@
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * A page for displaying Onboarding Content.
+ *
+ * @param onboardingPageState The page content that's displayed.
+ * @param onPrimaryButtonClick Invoked when the user clicks the primary button.
+ * @param onSecondaryButtonClick Invoked when the user clicks the secondary button.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun OnboardingPage(
+    onboardingPageState: OnboardingPageState,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier.semantics(mergeDescendants = true) {},
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.SpaceEvenly,
+    ) {
+        Image(
+            painter = painterResource(id = onboardingPageState.image),
+            contentDescription = null,
+        )
+
+        Column(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            Text(
+                text = stringResource(id = onboardingPageState.title),
+                color = FirefoxTheme.colors.textPrimary,
+                textAlign = TextAlign.Center,
+                style = FirefoxTheme.typography.headline5,
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Text(
+                text = stringResource(id = onboardingPageState.description),
+                color = FirefoxTheme.colors.textSecondary,
+                textAlign = TextAlign.Center,
+                style = FirefoxTheme.typography.body2,
+            )
+        }
+
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            PrimaryButton(
+                text = stringResource(id = onboardingPageState.primaryButtonText),
+                onClick = onPrimaryButtonClick,
+            )
+
+            if (onboardingPageState.secondaryButtonText != null) {
+                Spacer(modifier = Modifier.height(8.dp))
+                SecondaryButton(
+                    text = stringResource(id = onboardingPageState.secondaryButtonText),
```

Comment:
I think we can leave the named `id` parameter out of these `stringResource` calls

Reply:
I have discussed this with @MozillaNoah a long time ago, but I think it's fine to just leave these `id` in since they are autofilled when you autocomplete `stringResource` and I rather not have to force developers to delete it.